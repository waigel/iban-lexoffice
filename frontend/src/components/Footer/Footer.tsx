import {useTranslate} from '@tolgee/react';
import React from 'react';
import {LocalizationSelector} from '../LocalizationSelector/LocalizationSelector';

export const Footer = () => {
  const {t} = useTranslate();
  return (
    <footer className="fixed bottom-0 w-full text-sm text-gray-500">
      <div className="max-w-screen-xl py-4 mx-auto">
        <div className="flex flex-col md:flex-row justify-center gap-1 md:gap-4 text-center">
          <p>{t('COPYRIGHT_NOTICE')}</p>
          <a
            href={`${
              process.env.REACT_APP_API_ENDPOINT ?? 'http://localhost:8080'
            }`}
            target="_blank"
          >
            {t('FOOTER_LABEL_API_DOCU')}
          </a>
          <LocalizationSelector />
        </div>
      </div>
    </footer>
  );
};
