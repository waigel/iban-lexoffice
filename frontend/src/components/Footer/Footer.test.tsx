import React from 'react';
import TestRenderer from 'react-test-renderer';
import {Footer} from './Footer';

jest.mock('@tolgee/react', () => ({
  useTranslate: () => ({t: jest.fn(key => key), isLoading: false}),
}));

jest.mock('../LocalizationSelector/LocalizationSelector', () => ({
  LocalizationSelector: () => <div>LocalizationSelector</div>,
}));

it('should test the Footer component is rendered correctly', () => {
  const tree = TestRenderer.create(<Footer />);
  expect(tree.toJSON()).toMatchSnapshot();
});
