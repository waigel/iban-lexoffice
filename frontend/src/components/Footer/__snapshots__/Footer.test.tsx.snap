// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test the Footer component is rendered correctly 1`] = `
<footer
  className="fixed bottom-0 w-full text-sm text-gray-500"
>
  <div
    className="max-w-screen-xl py-4 mx-auto"
  >
    <div
      className="flex flex-col md:flex-row justify-center gap-1 md:gap-4 text-center"
    >
      <p>
        COPYRIGHT_NOTICE
      </p>
      <a
        href="undefined"
        target="_blank"
      >
        FOOTER_LABEL_API_DOCU
      </a>
      <div>
        LocalizationSelector
      </div>
    </div>
  </div>
</footer>
`;
